### PR TITLE
chore: Remove/replace deprecated networks

### PR DESCRIPTION
**Pull request by mikeshultz**
### What I did

- [Polygon's Mumbai replaced with Amoy](https://www.alchemy.com/blog/polygon-mumbai-testnet-deprecation)
- [Ethereum Goerli no longer supported](https://www.alchemy.com/blog/ethereum-goerli-testnet-deprecation)

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Comment by mikeshultz** [comment]
CI errors are confusing.  They don't occur for me locally.  Error messages suggest the plugins aren't getting installed, but according to the logs they are.

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -87,19 +87,15 @@ jobs:
...
             WEB3_ARBITRUM_SEPOLIA_ALCHEMY_API_KEY: ${{ secrets.WEB3_ARBITRUM_SEPOLIA_ALCHEMY_API_KEY }}
             WEB3_BASE_MAINNET_ALCHEMY_API_KEY: ${{ secrets.WEB3_BASE_MAINNET_ALCHEMY_API_KEY }}
-            WEB3_BASE_GOERLI_ALCHEMY_API_KEY: ${{ secrets.WEB3_BASE_GOERLI_ALCHEMY_API_KEY }}
             WEB3_BASE_SEPOLIA_ALCHEMY_API_KEY: ${{ secrets.WEB3_BASE_SEPOLIA_ALCHEMY_API_KEY }}
             WEB3_OPTIMISM_MAINNET_ALCHEMY_API_KEY: ${{ secrets.WEB3_OPTIMISM_MAINNET_ALCHEMY_API_KEY }}
-            WEB3_OPTIMISM_GOERLI_ALCHEMY_API_KEY: ${{ secrets.WEB3_OPTIMISM_GOERLI_ALCHEMY_API_KEY }}
             WEB3_OPTIMISM_SEPOLIA_ALCHEMY_API_KEY: ${{ secrets.WEB3_OPTIMISM_SEPOLIA_ALCHEMY_API_KEY }}
             WEB3_POLYGON_MAINNET_ALCHEMY_API_KEY: ${{ secrets.WEB3_POLYGON_MAINNET_ALCHEMY_API_KEY }}
-            WEB3_POLYGON_MUMBAI_ALCHEMY_API_KEY: ${{ secrets.WEB3_POLYGON_MUMBAI_ALCHEMY_API_KEY }}
+            WEB3_POLYGON_AMOY_ALCHEMY_API_KEY: ${{ secrets.WEB3_POLYGON_AMOY_ALCHEMY_API_KEY }}
```
[the remark this answers is not in the document]
Comment (mikeshultz) [reply .github/workflows/test.yaml:98]: @fubuloubu @antazoey  would one of you be able to handle this when you get a sec?

[the remark this answers is not in the document]
Comment (antazoey) [reply .github/workflows/test.yaml:98]: Added

**Review by antazoey — APPROVED** [review]
was a duplicate of https://github.com/ApeWorX/ape-alchemy/pull/65/files ?

**Comment by mikeshultz** [comment]
Got #65 merged and the only differences left are the removal of some Goerli-based testnets.

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**